account/src/main/java/com/whz/account/resource/StockAcquiredSubscriber.java

```diff
@@ -11,41 +11,78 @@
+ * messages containing acquired stocks. After receiving a message the "moneyInvested(double amount)"
+ * method will be triggered which matches the id of the acquired stock to our account and adds the
+ * invested value to our totalInvested value by creating a MoneyInvested event.
+ *
+ * @author Lation
+ */
 public enum StockAcquiredSubscriber {
   INSTANCE;
 
   private final Logger logger = Logger.basicLogger();
 
-  private static final String EXCHANGE_NAME = "logs"; // ???
+  private Stage stage;
 
+  /**
+   * Initializes and establishes a RabbitMQ connection, waits for messages, receives them and calls
+   * the appropriate moneyInvested() method in the Account Interface.
+   */
   StockAcquiredSubscriber() {
-    final ConnectionFactory factory = new ConnectionFactory();
-    factory.setHost("localhost");
+    String serviceName = System.getenv("RABBITMQ_SERVICE");
+    String exchangeName = System.getenv("RABBITMQ_EXCHANGE_STOCKACQUIRED");
+    String exchangeType = System.getenv("RABBITMQ_EXCHANGE_TYPE");
 
-    try (final Connection connection = factory.newConnection();
-        final Channel channel = connection.createChannel()) {
+    try {
+      final ConnectionFactory factory = new ConnectionFactory();
+      factory.setHost(serviceName);
+      final Connection connection = factory.newConnection();
 
-      channel.exchangeDeclare(EXCHANGE_NAME, "fanout");
+      final Channel channel = connection.createChannel();
+      channel.exchangeDeclare(exchangeName, exchangeType);
       String queueName = channel.queueDeclare().getQueue();
-      channel.queueBind(queueName, EXCHANGE_NAME, "");
+      channel.queueBind(queueName, exchangeName, "");
 
-      logger.debug("Waiting for messages..");
+      logger.debug("Started stock acquired subscriber");
+      logger.debug("Waiting for messages...");
 
       DeliverCallback deliverCallback =
           (consumerTag, delivery) -> {
             String message = new String(delivery.getBody(), StandardCharsets.UTF_8);
-            logger.debug("Received '" + message + "' with a length of: " + message.length());
+            StockData stockData = JsonSerialization.deserialized(message, StockData.class);
+
+            stage
+                .actorOf(
+                    Account.class,
+                    stage.addressFactory().from(stockData.owner),
+                    AccountEntity.class)
+                .andThenTo(
+                    account -> {
+                      return account.moneyInvested(stockData.amount);
+                    });
+
+            logger.debug("Stock acquired subscriber received message:'{}'", message);
           };
-      String test = channel.basicConsume(queueName, true, deliverCallback, consumerTag -> {});
 
-      logger.debug(test);
+      channel.basicConsume(queueName, true, deliverCallback, consumerTag -> {});
     } catch (IOException | TimeoutException e) {
       logger.debug(e.getMessage(), e);
```

Comment:
I think it is more useful to write a custom message here instead of repeating the error message of the exception.